File: WhatIfBudget/WhatIfBudget.Data/Models/InvestmentGoal.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WhatIfBudget.Data.Models
{
    public class InvestmentGoal : BaseEntity
    {
        public double AnnualReturnRate_Percent { get; set; } = 0;
        public UInt16 YearsToTarget { get; set; } = 0;
        public bool RolloverCompletedGoals { get; set; } = false;
        public double AdditionalBudgetAllocation { get; set; } = 0;

        public virtual Budget? Budget { get; set; }
    }
}

File: WhatIfBudget/WhatIfBudget.App/ClientApp/src/app/shared/models/investment-goal.ts
export class InvestmentGoal {
  id: number | undefined;
  yearsToTarget: number | undefined
  additionalBudgetAllocation: number | undefined;
  annualReturnRate_Percent: number | undefined;
  totalBalance: number | undefined
  rolloverCompletedGoals: boolean = false;
}

export class InvestmentTotals {
  addedDueToContribution: number | undefined;
  balanceAtTarget: number | undefined;
  totalInterestAccrued: number | undefined;
}
